guice-extension/src/main/java/name/falgout/jeffrey/testing/junit/guice/GuiceExtension.java

```diff
@@ -79,10 +84,44 @@ private static Injector createInjector(ExtensionContext context)
       InvocationTargetException {
     Optional<Injector> parentInjector = getParentInjector(context);
     List<? extends Module> modules = getNewModules(context);
-
+    if (isSharedInjector(context)) {
+      return createCachedInjector(modules, parentInjector);
+    }
     return parentInjector
-        .map(injector -> injector.createChildInjector(modules))
-        .orElseGet(() -> Guice.createInjector(modules));
+            .map(injector -> injector.createChildInjector(modules))
+            .orElseGet(() -> Guice.createInjector(modules));
+  }
+
+  private static Injector createCachedInjector(List<? extends Module> modules,
+      Optional<Injector> parentInjector) {
+    Set<? extends Class<?>> modulClass = modules.stream().map(Object::getClass).collect(toSet());
+    Optional<Injector> cachedInjector = getCachedInjector(modulClass);
+    if (cachedInjector.isPresent()) {
```

Comment:
Use INJECTOR_CACHE.computeIfAbsent instead of this conditional.

https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-